planner/core/expression_rewriter.go

```diff
@@ -1292,3 +1296,61 @@ func (er *expressionRewriter) toColumn(v *ast.ColumnName) {
 	}
 	er.err = ErrUnknownColumn.GenWithStackByArgs(v.String(), clauseMsg[er.b.curClause])
 }
+
+func (er *expressionRewriter) evalDefaultExpr(v *ast.DefaultExpr) {
+	stkLen := len(er.ctxStack)
+	col, ok := er.ctxStack[stkLen-1].(*expression.Column)
+	if !ok {
+		col, er.err = er.schema.FindColumn(v.Name)
+		if er.err != nil {
+			return
+		}
+	}
+
+	table, err := er.b.is.TableByName(col.DBName, col.OrigTblName)
+	if err != nil {
+		er.err = errors.Trace(err)
```

Comment:
Once it's traced, no need to trace it anymore.

Reply:
@winoros thanks for the review, but I don't get your point... when you talk about 'traced', do you mean if the error is `HasStack`?

Reply:
He means you can just use `er.err = err`, no need to call `errors.Trace()`

Reply:
yes.
```
// Trace just calls AddStack.
func Trace(err error) error {
	return AddStack(err)
}
```

Reply:
how do we know if the error is traced already?

Reply:
Is it because we've used stack to generate the error?
https://github.com/pingcap/tidb/blob/0c8e967858f4abdc0d2f24b4fdd167fbcbfcb356/infoschema/infoschema.go#L177

Reply:
Yes